core/src/main/scala/sttp/client/testing/SttpBackendStub.scala

```diff
@@ -97,49 +120,75 @@ class SttpBackendStub[F[_], S](
   override def responseMonad: MonadError[F] = monad
 
   class WhenRequest(p: Request[_, _] => Boolean) {
-    def thenRespondOk(): SttpBackendStub[F, S] =
+    def thenRespondOk(): SttpBackendStub[F, S, WS_HANDLER] =
       thenRespondWithCode(StatusCode.Ok)
-    def thenRespondNotFound(): SttpBackendStub[F, S] =
+    def thenRespondNotFound(): SttpBackendStub[F, S, WS_HANDLER] =
       thenRespondWithCode(StatusCode.NotFound, "Not found")
-    def thenRespondServerError(): SttpBackendStub[F, S] =
+    def thenRespondServerError(): SttpBackendStub[F, S, WS_HANDLER] =
       thenRespondWithCode(StatusCode.InternalServerError, "Internal server error")
-    def thenRespondWithCode(status: StatusCode, msg: String = ""): SttpBackendStub[F, S] = {
+    def thenRespondWithCode(status: StatusCode, msg: String = ""): SttpBackendStub[F, S, WS_HANDLER] = {
       thenRespond(Response(msg, status, msg))
     }
-    def thenRespond[T](body: T): SttpBackendStub[F, S] =
+    def thenRespond[T](body: T): SttpBackendStub[F, S, WS_HANDLER] =
       thenRespond(Response[T](body, StatusCode.Ok, "OK"))
-    def thenRespond[T](resp: => Response[T]): SttpBackendStub[F, S] = {
+    def thenRespond[T](resp: => Response[T]): SttpBackendStub[F, S, WS_HANDLER] = {
       val m: PartialFunction[Request[_, _], F[Response[_]]] = {
         case r if p(r) => monad.eval(resp)
       }
-      new SttpBackendStub(monad, matchers.orElse(m), fallback)
+      new SttpBackendStub[F, S, WS_HANDLER](monad, matchers.orElse(m), wsMatchers, fallback)
+    }
+
+    /**
+      * When [[openWebsocket()]] is called, it will ignore handler and return given result.
+      * It is intended to be used when [[openWebsocket()]] is called with sttp supplied handlers
+      * that return wrapped WebSocket and WebSocketStub as WS_RESULT.
+      * */
+    def thenWebsocketResponse[WS_RESULT](headers: Headers, result: WS_RESULT) = {
```

Comment:
maybe: `.thenWebSocketResult` and `.thenWebSocketResultFromHandler`? To keep the result/handler terminology